### PR TITLE
fix(array): add element size validation and overflow checks

### PR DESCRIPTION
**Pull request by Karticoder**
While working with the lv_array module in the codebase, I noticed lv_array_concat lacked defensive checks present in other array functions. It could dereference NULL pointers, silently corrupt data when concatenating arrays with different element sizes, or trigger integer overflows on size addition. I added LV_ASSERT debug macros, runtime checks with error logging, integer overflow protection, updated the header documentation, and expanded the unit test suite to cover these edge cases.

<a href="https://cubic.dev/pr/lvgl/lvgl/pull/10366?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by halyssonJr** [comment]
@Karticoder, you made some great improvements. Please keep contributing.

File: src/misc/lv_array.c [hunk 1]
```diff
@@ -209,14 +209,14 @@ lv_result_t lv_array_erase(lv_array_t * array, uint32_t start, uint32_t end)
 
 bool lv_array_resize(lv_array_t * array, uint32_t new_capacity)
 {
+    if(array == NULL) return false;
```
Comment (AndreCostaaa) [inline src/misc/lv_array.c]: Assert?

Comment (kisvegabor) [reply src/misc/lv_array.c]: Done. Other asserts are missing too, but these are unrelated.

**Review by AndreCostaaa — APPROVED** [review]
